platform/app/cypress/support/commands.js

```diff
@@ -198,54 +200,24 @@ Cypress.Commands.add('expectMinimumThumbnails', (seriesToWait = 1) => {
 });
 
 //Command to wait DICOM image to load into the viewport
-Cypress.Commands.add('waitDicomImage', (timeout = 50000) => {
-  const loaded = cy.isPageLoaded();
-
-  if (loaded) {
-    cy.window()
-      .its('cornerstone')
-      .then({ timeout }, $cornerstone => {
-        return new Cypress.Promise(resolve => {
-          const onEvent = renderedEvt => {
-            const element = renderedEvt.detail.element;
-
-            element.removeEventListener(
-              $cornerstone.Enums.Events.IMAGE_RENDERED,
-              onEvent
-            );
-            $cornerstone.eventTarget.removeEventListener(
-              $cornerstone.Enums.Events.IMAGE_RENDERED,
-              onEvent
-            );
-            resolve();
-          };
-          const onEnabled = enabledEvt => {
-            const element = enabledEvt.detail.element;
-
-            element.addEventListener(
-              $cornerstone.Enums.Events.IMAGE_RENDERED,
-              onEvent
-            );
-
-            $cornerstone.eventTarget.removeEventListener(
-              $cornerstone.Enums.Events.ELEMENT_ENABLED,
-              onEnabled
-            );
-          };
-          const enabledElements = $cornerstone.getEnabledElements();
-          if (enabledElements && enabledElements.length) {
-            // Sometimes the page finishes rendering before this gets run,
-            // if so, just resolve immediately.
-            resolve();
-          } else {
-            $cornerstone.eventTarget.addEventListener(
-              $cornerstone.Enums.Events.ELEMENT_ENABLED,
-              onEnabled
+Cypress.Commands.add('waitDicomImage', () => {
+  cy.window()
+    .its('cornerstone')
+    .should($cornerstone => {
+      const enabled = $cornerstone.getEnabledElements();
+      if (enabled?.length) {
+        enabled.forEach((item, i) => {
+          if (item.viewport.viewportStatus !== 'rendered') {
```

Comment:
we should grab 'rendered' from Enums

Reply:
App doesn't include @cornerstonejs/core as that is part of the extension.  I'd rather not have the dependency at that level, so that kind of leaves directly encoding the value.  If you prefer me to add the dependency then I can, but I really feel all cornerstone dependencies should be in extensions.

Reply:
you are already using it in line 207, $cornerstone